### PR TITLE
add cos bucket functionality

### PR DESCRIPTION
**Pull request by n-marton**

### CODE REVIEW

**Comment by hkantare** [comment]
Hi.. Thanks for raising PR . We are glad for your support for contributing towards opensource.

I have few concerns

* In resource the auth endpoint is hardcoded and its prod endpoint..The terraform-provider-ibm can be used on stage , prod . When we run on stage environment it may not be proper endpoint.

* In resource you are passing apikey to sdk..Can we add support for token(IAM token). The terraform-provider-ibm is not part of schematics service (https://cloud.ibm.com/schematics/landing) .The user can provision resources without passing an APIkey

**Comment by n-marton** [comment]
Hi,

Done & Done

File: ibm/resource_ibm_cos_bucket.go [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+				ForceNew:     true,
+			},
+			"s3_endpoint_public": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "CRN of resource instance",
+			},
+			"s3_endpoint_private": {
+				Type:        schema.TypeString,
+				Computed:    true,
```
Comment (hkantare) [inline ibm/resource_ibm_cos_bucket.go]: Can the description can be changed

File: ibm/resource_ibm_cos_bucket.go [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+			"storage_class": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAllowedStringValue(storageClass),
+				ForceNew:     true,
+			},
+			"s3_endpoint_public": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "CRN of resource instance",
```
Comment (hkantare) [inline ibm/resource_ibm_cos_bucket.go]: Can the description can be changed

**Pull request merged**